metadata-ingestion/docs/transformer/intro.md

```diff
@@ -0,0 +1,18 @@
+---
+title: "Introduction"
+---
+
+# Transformers
+
+## What’s a transformer?
+
+Oftentimes we want to modify metadata before it reaches the ingestion sink – for instance, we might want to add custom tags, ownership, properties, or patch some fields. A transformer allows us to do exactly these things.
+
+Moreover, a transformer allows one to have fine-grained control over the metadata that’s ingested without having to modify the ingestion framework's code yourself. Instead, you can write your own module that can transform metadata events however you like. To include a transformer into a recipe, all that's needed is the name of the transformer as well as any configuration that the transformer needs.
+
+## Provided transformers
+
+Aside from the option of writing your own transformer (see below), we provide some simple transformers for the use cases of adding: tags, glossary terms, properties and ownership information.
+
+DataHub provided transformers are 
+- [Dataset Transformer](./dataset_transformer.md)
```

Comment:
might want to make it clear that there's many transformers listed under this doc, not just one

Reply:
@mohdsiddique actually for the transformer docs, can we generate one .md file per transformer / set of related transformers rather than having everything in metadata-ingestion/docs/transformer/dataset_transformer.md?

Reply:
> might want to make it clear that there's many transformers listed under this doc, not just one
All transformers are working on dataset entity only, hene I created a dataset_transformer.md. and on dataset_transformer.md we have a complete table of transformers. the current refactoring is much better than the existing transformer.md file.

Reply:
> @mohdsiddique actually for the transformer docs, can we generate one .md file per transformer / set of related transformers rather than having everything in metadata-ingestion/docs/transformer/dataset_transformer.md?

We wanted to have a single transformer that will work on all entities to apply tag, ownership, glossaryTerm, and domain.  so if we create per transformer readme file then we might need to revert again to a single file later.